packages/near-membrane-dom/src/browser-realm.ts

```diff
@@ -154,18 +211,6 @@ export default function createVirtualEnvironment(
     // We intentionally skip remapping Window.prototype because there is nothing
     // in it that needs to be remapped.
     env.lazyRemap(blueRefs.EventTargetProto, blueRefs.EventTargetProtoOwnKeys);
```

Comment:
don't need to do this for ShadowRealm since the globalThis.__proto__ remapping covers this.

Reply:
same applies to `removeWindowDescriptors`, but in that case I'm not sure about `chrome` removal.

Reply:
same for `unforgeablePoisonedWindowKeys`... which is not needed, and adding it cause other issues in shadow realm.

Reply:
additionally, 

```
    // window.__proto__ (aka Window.prototype)
    // window.__proto__.__proto__ (aka WindowProperties.prototype)
    // window.__proto__.__proto__.__proto__ (aka EventTarget.prototype)
    env.link('__proto__', '__proto__', '__proto__');
    env.remapProto(blueRefs.document, blueRefs.DocumentProto);
``` 

is not needed, instead, you need to set the __proto__ of the `globalThis` in the shadow realm to a proxy of a `window.__proto__` on the main window, that way, the rest of this is rendered useless since the proto is remapped entirely. This poses the problem of potentially leaking `globalThis[0]` into the shadow realm, so, very likely, a distortion is needed for that part of the proto chain to avoid leaking values.

Reply:
There might be other cases that are also forking logic between the two worlds.

Reply:
I'll work on these changes today